apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -79,3 +444,62 @@ def _from_table(self, arrow_table: pa.Table) -> pa.SparseCOOTensor:
             ]
         ).T
         return pa.SparseCOOTensor.from_numpy(coo_data, coo_coords, shape=self.shape)
+
+
+def _arrow_table_reader(sr: clib.SOMAArray) -> Iterator[pa.Table]:
+    """Private. Simple Table iterator on any Array"""
+    tbl = sr.read_next()
+    while tbl is not None:
+        yield tbl
+        tbl = sr.read_next()
+
+
+def _coords_strider(
+    coords: options.SparseNDCoord, length: int, stride: int
+) -> Iterator[npt.NDArray[np.int64]]:
+    """
+    Private.
+
+    Iterate over major coordinates, in stride sized steps, materializing each step as an
+    ndarray of coordinate values. Will be sorted in ascending order.
+
+    NB: SOMA slices are _closed_ (i.e., inclusive of both range start and stop)
+    """
+    # normalize coord to either a slice or ndarray
+    if coords is None or (isinstance(coords, slice) and coords == slice(None)):
```

Comment:
probably no need to `isinstance` here; `coords == slice(None)` is going to be False for non-slices (I would expect)

Reply:
Acutally, it is needed to handle the case where coords is an NDArray and will broadcast the equality test along the entire array.

Reply:
I see. Could you add a comment to this effect? (not back in the full review mode right now; if you already have that's good)

Reply:
will do